### PR TITLE
Added cypress e2e test and loader unit test for nl-search

### PR DESCRIPTION
**Pull request by TheRBajaj**
- Cypress test tests the e2e search functionality when nl-search is toggled on.
- NLLoader tests if the loader works as intended

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: cypress/e2e/userIsAbleToSearch.cy.js [hunk 1]
```diff
@@ -62,4 +62,15 @@ describe("Search", () => {
...
 		cy.get("div").should("contain", "Failed to search place by text, 'Text' must have length at least 1");
 	});
+
+	it("should enable nl search and allow user to view POI after nl search", { scrollBehavior: false }, () => {
+		cy.get('[class="amplify-label amplify-switch__wrapper amplify-switch__wrapper--end"]').click();
+		cy.get('[placeholder="Search"]').click();
+		cy.wait(2000);
+		cy.get('[inputmode="search"]').type("Show me three pizza places near Eiffel Tower?").wait(5000).type("{enter}");
+		cy.wait(15000);
+		cy.get("div").should('contain', ["Pizza Fiorentina"]);
```
Comment (marinrelatic) [inline cypress/e2e/userIsAbleToSearch.cy.js:72]: We should not assert data returned from data partners. We don't control what data partner returns. This data can change and result in valid integration test failure (flaky tests)

**Review by marinrelatic — CHANGES_REQUESTED** [review]
Result assertion requires change, otherwise looks good. 
Can we have a test to assert NL places search is not available if the env variables are not set?

File: cypress/e2e/userIsAbleToSearch.cy.js [hunk 2]
```diff
@@ -62,4 +62,15 @@ describe("Search", () => {
 		cy.wait(2000);
 		cy.get("div").should("contain", "Failed to search place by text, 'Text' must have length at least 1");
 	});
+
+	it("should enable nl search and allow user to view POI after nl search", { scrollBehavior: false }, () => {
+		cy.get('[class="amplify-label amplify-switch__wrapper amplify-switch__wrapper--end"]').click();
```
Comment (thehoneymad) [inline cypress/e2e/userIsAbleToSearch.cy.js:67]: Can we have an id for ths?

File: cypress/e2e/userIsAbleToSearch.cy.js [hunk 3]
```diff
@@ -62,4 +62,15 @@ describe("Search", () => {
...
+		cy.get('[placeholder="Search"]').click();
+		cy.wait(2000);
+		cy.get('[inputmode="search"]').type("Show me three pizza places near Eiffel Tower?").wait(5000).type("{enter}");
+		cy.wait(15000);
```
Comment (thehoneymad) [inline cypress/e2e/userIsAbleToSearch.cy.js:71]: Is there a way we can be a little nicer here?

As in can we wait up to 15 seconds maximum for results to show up over always waiting 15 seconds.

Comment (TheRBajaj) [reply cypress/e2e/userIsAbleToSearch.cy.js:71]: Couldn't find an explicit way to do this, but will update it later when I find something that will result in that functionality.

File: src/atomicui/organisms/AuthGeofenceBox/AuthGeofenceBox.tsx [hunk 4]
```diff
@@ -126,7 +126,8 @@ const AuthGeofenceBox: React.FC<AuthGeofenceBoxProps> = ({ mapRef, setShowAuthGe
 						exact,
 						sg => setSuggestions(sg),
 						TriggeredByEnum.GEOFENCE_MODULE,
-						AnalyticsEventActionsEnum.AUTOCOMPLETE
+						AnalyticsEventActionsEnum.AUTOCOMPLETE,
+						false
```
Comment (marinrelatic) [inline src/atomicui/organisms/AuthGeofenceBox/AuthGeofenceBox.tsx:130]: What is the purpose of this change?

File: src/atomicui/organisms/RouteBox/RouteBox.tsx [hunk 5]
```diff
@@ -268,7 +268,8 @@ const RouteBox: React.FC<RouteBoxProps> = ({ mapRef, setShowRouteBox, isSideMenu
 								: setSuggestions({ ...suggestions, to: sg });
 						},
 						TriggeredByEnum.ROUTE_MODULE,
-						action
+						action,
+						false
```
Comment (marinrelatic) [inline src/atomicui/organisms/RouteBox/RouteBox.tsx:272]: Same as comment above. What is the purpose of this change?

File: src/atomicui/organisms/SearchBox/SearchBox.tsx [hunk 6]
```diff
@@ -437,9 +433,11 @@ const SearchBox: React.FC<SearchBoxProps> = ({
 						}
 						crossOrigin={undefined}
 					/>
-					{NL_BASE_URL && NL_API_KEY ? (
+					{NL_BASE_URL && NL_API_KEY && currentMapProvider === MapProviderEnum.ESRI ? (
```
Comment (marinrelatic) [inline src/atomicui/organisms/SearchBox/SearchBox.tsx:436]: I would suggest to avoid putting non related changes into same commit

File: src/atomicui/organisms/AuthGeofenceBox/AuthGeofenceBox.tsx (hunk 4 above)
Comment (TheRBajaj) [reply src/atomicui/organisms/AuthGeofenceBox/AuthGeofenceBox.tsx]: removed it, set a default value for that parameter

File: src/atomicui/organisms/RouteBox/RouteBox.tsx (hunk 5 above)
Comment (TheRBajaj) [reply src/atomicui/organisms/RouteBox/RouteBox.tsx]: removed it, set a default value for that parameter

**Review by marinrelatic — APPROVED** [review]
